controllers/nodehealthcheck_controller.go

```diff
@@ -412,6 +436,26 @@ func (r *NodeHealthCheckReconciler) remediate(ctx context.Context, node *v1.Node
 	return nextReconcile, nil
 }
 
+func (r *NodeHealthCheckReconciler) isControlPlaneRemediationAllowed(node *v1.Node, template *unstructured.Unstructured) (bool, error) {
+	if !utils.IsControlPlane(node) {
+		return true, nil
+	}
+	// check all remediation CRs. If there already is one for another control plane node, skip remediation
+	remediations, err := r.getAllInflightRemediations(template)
```

Comment:
Nit: I suggest returning here:
remediation.GetName() == node.GetName(), nil instead of continue iterating